core/parachain/availability/recovery/recovery_impl.cpp

```diff
@@ -31,15 +31,16 @@ namespace {
 
   const std::array<std::string, 3> results = {"success", "failure", "invalid"};
 
+  // NOLINTNEXTLINE(cppcoreguidelines-macro-usage)
 #define incFullRecoveriesFinished(strategy, result)                          \
-  do {                                                                       \
+  [&] {                                                                      \
```

Comment:
`do { ... } while (false)` was used for correct semicolon after statement.
https://gcc.gnu.org/onlinedocs/cpp/Swallowing-the-Semicolon.html

Reply:
Call temporary lambda does the same, but pass clang-tidy rule `cppcoreguidelines-avoid-do-while`